pkg/skaffold/kubernetes/manifest/images.go

```diff
@@ -200,6 +200,12 @@ func (r *imageReplacer) Visit(gk apimachinery.GroupKind, navpath string, o map[s
 		log.Entry(context.TODO()).Debugf("Couldn't parse image [%s]: %s", image, err.Error())
 		return false
 	}
+	// this is a hack to properly support `imageStrategy=helm+explicitRegistry` from Skaffold v1.X.X
```

Comment:
can you add a bit more detail about this hack? Like with an example of `image` value, and why we're setting `BaseName` to `Repo`.

Reply:
is it possible that any other regular scenario will conflict with this conditional?

Reply:
I don't believe there is realistically any regular scenario where this will conflict.  I have tried with both artifact/artifactOverrides names of the format:
```
  artifacts:
  - image: gcr.io/aprindle-test-cluster/skaffold-helm
```
```
  artifacts:
  - image: skaffold-helm
```
and this works for both cases

Reply:
Updated the comment to have more explanation:

```
	// this is a hack to properly support `imageStrategy=helm+explicitRegistry` from Skaffold v1.X.X which has the form:
	// image: "{{.Values.image.registry}}/{{.Values.image.repository}}:{{.Values.image.tag}}"
	// works by looking for intermediate helm replacement of the form image: <artifactName>/<artifactName>:<artifactName>
	// and treating that as just <artifact> by modifying the parsed representation.
```